Cross build of etcd container doesn't seem to work

Am I doing something wrong?

```
$ make TAG=2.3.7 ARCH=arm push
# Copy the content in this dir to the temp dir,
# without copying the subdirectories.
find ./ -maxdepth 1 -type f | xargs cp -t /tmp/tmp.1XlBL7nThG
make -C ../../../ WHAT=cluster/images/etcd/attachlease KUBE_STATIC_OVERRIDES="etcd";
make[1]: Entering directory `/usr/local/google/home/dbsmith/code/k2/src/k8s.io/kubernetes'
make[2]: Entering directory `/usr/local/google/home/dbsmith/code/k2/src/k8s.io/kubernetes'
make[2]: Leaving directory `/usr/local/google/home/dbsmith/code/k2/src/k8s.io/kubernetes'
+++ [0908 15:32:54] Generating bindata:
    /usr/local/google/home/dbsmith/code/k2/src/k8s.io/kubernetes/test/e2e/framework/gobindata_util.go
+++ [0908 15:32:56] Building the toolchain targets:
    k8s.io/kubernetes/hack/cmd/teststale
+++ [0908 15:32:57] Building go targets for linux/amd64:
    cluster/images/etcd/attachlease
make[1]: Leaving directory `/usr/local/google/home/dbsmith/code/k2/src/k8s.io/kubernetes'
cp "../../../_output/bin/attachlease" /tmp/tmp.1XlBL7nThG
# Download etcd in a golang container and cross-compile it statically
docker run -it -v /tmp/tmp.1XlBL7nThG:/etcdbin golang:1.6.3 /bin/bash -c \
                "git clone https://github.com/coreos/etcd /go/src/github.com/coreos/etcd \
                && cd /go/src/github.com/coreos/etcd \
                && git checkout v2.3.7 \
                && GOARM=6 GOARCH=arm ./build \
                && cp bin/arm/* /etcdbin"
Cloning into '/go/src/github.com/coreos/etcd'...
remote: Counting objects: 52271, done.
remote: Compressing objects: 100% (23/23), done.
remote: Total 52271 (delta 6), reused 0 (delta 0), pack-reused 52248
Receiving objects: 100% (52271/52271), 22.78 MiB | 789.00 KiB/s, done.
Resolving deltas: 100% (32803/32803), done.
Checking connectivity... done.
Note: checking out 'v2.3.7'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b new_branch_name

HEAD is now at fd17c91... version: bump to v2.3.7
cp: cannot stat 'bin/arm/*': No such file or directory
make: *** [build] Error 1
```

--- a/cluster/images/etcd/Makefile
+++ b/cluster/images/etcd/Makefile
@@ -15,7 +15,7 @@
 # Build the etcd image
 #
 # Usage:
-# 	[TAG=2.2.1] [REGISTRY=gcr.io/google_containers] [ARCH=amd64] [BASEIMAGE=busybox] make (build|push)
+# 	[TAG=3.0.4] [REGISTRY=gcr.io/google_containers] [ARCH=amd64] [BASEIMAGE=busybox] make (build|push)
 
 TAG?=3.0.4
 ARCH?=amd64
@@ -42,8 +42,9 @@ build:
 	# without copying the subdirectories.
 	find ./ -maxdepth 1 -type f | xargs cp -t $(TEMP_DIR)
 
-	make -C ../../../ WHAT=cluster/images/etcd/attachlease KUBE_STATIC_OVERRIDES="etcd";
-	cp "../../../_output/bin/attachlease" $(TEMP_DIR)
+	# Compile attachlease
+	docker run -it -v $(shell pwd)/../../../:/go/src/k8s.io/kubernetes -v $(TEMP_DIR):/build -e GOARCH=$(ARCH) golang:$(GOLANG_VERSION) \
+		/bin/bash -c "CGO_ENABLED=0 go build -o /build/attachlease k8s.io/kubernetes/cluster/images/etcd/attachlease"
 
 ifeq ($(ARCH),amd64)
 
@@ -57,7 +58,7 @@ else
 		&& cd /go/src/github.com/coreos/etcd \
 		&& git checkout v$(TAG) \
 		&& GOARM=$(GOARM) GOARCH=$(ARCH) ./build \
-		&& cp bin/$(ARCH)/* /etcdbin"
+		&& cp -f bin/$(ARCH)/etcd* bin/etcd* /etcdbin; echo 'done'"
 
 	# Add this ENV variable in order to workaround an unsupported arch blocker
 	# The multiarch feature is in an limited and experimental state right now, and etcd should work fine on arm64

